src/main/drivers/max7456.c

```diff
@@ -0,0 +1,247 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "common/printf.h"
+#include "platform.h"
+#include "version.h"
+
+#ifdef USE_MAX7456
+
+#include "drivers/bus_spi.h"
+#include "drivers/light_led.h"
+#include "drivers/system.h"
+
+#include "max7456.h"
+#include "max7456_symbols.h"
+
+#define DISABLE_MAX7456       IOHi(max7456CsPin)
+#define ENABLE_MAX7456        IOLo(max7456CsPin)
+
+static IO_t max7456CsPin = IO_NONE;
+
+/** Artificial Horizon limits **/
+#define AHIPITCHMAX 200             // Specify maximum AHI pitch value displayed. Default 200 = 20.0 degrees
+#define AHIROLLMAX  400             // Specify maximum AHI roll value displayed. Default 400 = 40.0 degrees
+#define AHISIDEBARWIDTHPOSITION 7
+#define AHISIDEBARHEIGHTPOSITION 3
+
+uint16_t max_screen_size;
+uint8_t  video_signal_type = 0;
+uint8_t  max7456_lock = 0;
+char max7456_screen[VIDEO_BUFFER_CHARS_PAL];
+
+
+uint8_t max7456_send(uint8_t add, uint8_t data) {
+    spiTransferByte(MAX7456_SPI_INSTANCE, add);
+    return spiTransferByte(MAX7456_SPI_INSTANCE, data);
+}
+
+
+void max7456_init(uint8_t video_system) {
+    uint8_t max_screen_rows;
+    uint8_t srdata = 0;
+    uint16_t x;
+    char buf[LINE];
+
+#ifdef MAX7456_SPI_CS_PIN
+    max7456CsPin = IOGetByTag(IO_TAG(MAX7456_SPI_CS_PIN));
+#endif
+    IOInit(max7456CsPin, OWNER_SYSTEM, RESOURCE_SPI);
+    IOConfigGPIO(max7456CsPin, SPI_IO_CS_CFG);
+
+    //Minimum spi clock period for max7456 is 100ns (10Mhz)
+    spiSetDivisor(MAX7456_SPI_INSTANCE, SPI_9MHZ_CLOCK_DIVIDER);
+
+    delay(1000);
+    // force soft reset on Max7456
+    ENABLE_MAX7456;
+    max7456_send(VM0_REG, MAX7456_RESET);
+    delay(100);
+
+    srdata = max7456_send(0xA0, 0xFF);
+    if ((0x01 & srdata) == 0x01){     //PAL
+          video_signal_type = VIDEO_MODE_PAL;
+    }
+    else if((0x02 & srdata) == 0x02){ //NTSC
+        video_signal_type = VIDEO_MODE_NTSC;
+    }
+
+    // Override detected type: 0-AUTO, 1-PAL, 2-NTSC
+    switch(video_system) {
+        case PAL:
+            video_signal_type = VIDEO_MODE_PAL;
+            break;
+        case NTSC:
+            video_signal_type = VIDEO_MODE_NTSC;
+            break;
+    }
+
+    if (video_signal_type) { //PAL
+        max_screen_size = VIDEO_BUFFER_CHARS_PAL;
+        max_screen_rows = VIDEO_LINES_PAL;
+    } else {                 // NTSC
+        max_screen_size = VIDEO_BUFFER_CHARS_NTSC;
+        max_screen_rows = VIDEO_LINES_NTSC;
+    }
+
+    // set all rows to same charactor black/white level
+    for(x = 0; x < max_screen_rows; x++) {
+        max7456_send(MAX7456ADD_RB0 + x, BWBRIGHTNESS);
+    }
+
+    // make sure the Max7456 is enabled
+    max7456_send(VM0_REG, OSD_ENABLE | video_signal_type);
+
+    DISABLE_MAX7456;
+    delay(100);
+
+    // display logo
```

Comment:
Display of logo does not belong in the device driver - it should be in the IO layer, ie `osdInit`.